crates/rpc/rpc/src/debug.rs

```diff
@@ -646,9 +649,14 @@ where
                         .or_insert_with(|| HashedStorage::new(account.status.was_destroyed()));
 
                     if let Some(account) = account.account {
+                        state_preimages.insert(hashed_address, alloy_rlp::encode(address));
+
                         for (slot, value) in account.storage {
-                            let hashed_slot = keccak256(B256::from(slot));
+                            let slot = B256::from(slot);
+                            let hashed_slot = keccak256(slot);
                             storage.storage.insert(hashed_slot, value);
+
+                            state_preimages.insert(hashed_slot, alloy_rlp::encode(slot));
```

Comment:
ideally i'd want to reuse `HashedPostState::from_cache_state` for this, but no need to tackle here

Reply:
right, but you'd need to walk the cache state anyway because `HashedPostState` won't contain pre-images, no?